specs/transport/car/carv2/index.md

```diff
@@ -0,0 +1,177 @@
+
+1. **Bits 0 to 7**: read as an unsigned little-endian 8-bit integer describing the ordering of the blocks in the CARv1 payload:
+   * `0` indicates _no particular order_, no assumptions should be made regarding determinism of payload layout.
+   * `1` indicates a _depth-first_ ordering of blocks in the DAG(s) referenced by the "roots" array. (**TODO:** "depth-first" isn't enough to be clear about what this means for a DAG walk - e.g. do we yield the current block before going to the next, or wait till we reach the depth before yielding blocks?)
+   * `2` indicates a _breadth-first_ ordering of blocks in the DAG(s) referenced by the "roots" array. (**TODO:** ditto ^ what does this specifically mean in DAG terms)
+   * `3` indicates that the ordering is dictated by an [IPLD Selector](../specs/selectors/index.md) which itself is serialized into the CARv1 payload and referenced in the last element in the "roots" array. i.e. deserializing the block referenced by the last element of the roots array in the CARv1 header should yield a data model form of the IPLD selector that was used to select and order the blocks in the CARv1 payload.
+   * `4` has the same meaning as `3` except that the IPLD selector is encoded as a DAG-CBOR IPLD block immediately following the Header and is prefixed with a varint indicating the length of the bytes to decode.
+2. **Bit 8** indicating whether the blocks have been properly deduplicated (`1`) and that a reader should not expect any repeated CIDs, or that the blocks _may_ contain duplicates (`0`) (perhaps as a trade-off for speed or memory consumption during generation).
+
+Future amendments of this specification may introduce additional characteristics, including:
+
+* **TODO:** what else are we potentially filling 128-bits up with?
+
+### CARv1 data payload
+
+The CARv1 data payload starts at the "Data offset" indicated by the CARv2 header. It is important that a decoder adhere to this offset rather than assuming the data payload begins immediately after the header in order to make allowance for additional data following the header as indicated by the characteristics field in future amendments to this specification.
+
+The CARv1 data payload ends after the number of bytes indicated by the "Data length" field in the header. This is important as the CARv1 parser has no means to determine the end of the container other than encountering the end of a byte stream. Therefore, a CARv2 decoder may defer to a CARv1 decoder to load the `CID:Bytes` sequence payload and "roots" array, but must be able to deliver it a byte stream that strictly begins at "Data offset" and ends at "Data length".
+
+The CARv1 data payload must include a valid CARv1 header, including a roots array and a `version` field with a value strictly of `1`.
```

Comment:
```suggestion
The data payload is a complete, self-contained CAR v1 archive. As such, it must include a valid CARv1 header, including a roots array and a `version` field with a value strictly of `1` followed by a series of data blocks.
```

I also wonder if we want to say a sentence about why. e.g. "This allows conversion of a CAR v2 to a CAR v1 without any further processing."

Reply:
:thumbsup: